### ISSUE TITLE
JSDoc for functions deconstructed renamed params

### ISSUE THREAD

**Issue by joelpro2** [gajus/eslint-plugin-jsdoc#776]
Don't know if this is a bug or a question

given a function with multiple deconstructed params how should I do it on the doc?

```javascript
/**
 * @param {Store}          context
 * @param {Store.commit}   context.commit
 * @param {Store.getters}  context.getters
 * @param {Store.dispatch} context.dispatch
 * @param {object}         payload
 * @param {object}         payload.new
 * @param {string}         payload.libraryType
 */
const updateStyleVersion = async (
  { commit, getters, dispatch },
  { new: newPayload, libraryType }
) => {
```

`valid-types` returns the following warning
`Syntax error in namepath: payload.new`

` * @param {object}         payload.newPayload:payload.new`
also throws a warning for check-param-names
` @param "payload.newPayload:payload.new" does not exist on payload`

**Comment by brettz9** [comment]
You can currently wrap such keywords in quotation marks. So you can do:

```js
* @param {object}         payload."new"
```

@simonseyock : Does this seem to be working as should be expected? i.e., does JSDoc expect keywords to be escaped even in this context?

**Comment by joelpro2** [comment]
And there's any way to document the alias (payload.newPayload)?

**Comment by brettz9** [comment]
> And there's any way to document the alias (payload.newPayload)?

No, JSDoc doesn't provide any formal mechanism for doing so. But in viewing the JSDoc block as the public API docs for a given function, I personally tend not to miss it, though one could put it in the description.

**Comment by simonseyock** [comment]
> You can currently wrap such keywords in quotation marks. So you can do:
> 
> ```js
> * @param {object}         payload."new"
> ```
> 
> @simonseyock : Does this seem to be working as should be expected? i.e., does JSDoc expect keywords to be escaped even in this context?

I don't think so. I have to test this, but it could that this could be considered a bug.
